api/handlers/space.go

```diff
@@ -148,6 +151,36 @@ func (h *Space) get(r *http.Request) (*routing.Response, error) {
 	return routing.NewResponse(http.StatusOK).WithBody(presenter.ForSpace(space, h.apiBaseURL)), nil
 }
 
+func (h *Space) deleteRoutes(r *http.Request) (*routing.Response, error) {
+	authInfo, _ := authorization.InfoFromContext(r.Context())
+	logger := logr.FromContextOrDiscard(r.Context()).WithName("handlers.space.delete-unmapped-routes")
+
+	spaceDeleteRoutes := new(payloads.SpaceDeleteRoutes)
+	if err := h.requestValidator.DecodeAndValidateURLValues(r, spaceDeleteRoutes); err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "Failed to decode and validate request values")
+	}
+
+	spaceGUID := routing.URLParam(r, "guid")
+	_, err := h.spaceRepo.GetSpace(r.Context(), authInfo, spaceGUID)
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, apierrors.ForbiddenAsNotFound(err), "Failed to fetch space", "spaceGUID", spaceGUID)
+	}
+
+	routes, err := h.routeRepo.ListRoutes(r.Context(), authInfo, repositories.ListRoutesMessage{SpaceGUIDs: []string{spaceGUID}, IsUnmapped: true})
+	if err != nil {
+		return nil, apierrors.LogAndReturn(logger, err, "Failed to delete unmapped routes for space", "SpaceGUID", spaceGUID)
+	}
+
+	for _, route := range routes {
```

Comment:
I am wondering if we should push the deletion to the repository to simplify the handler. It wouldn't change much, don't have a hard opinion...

Reply:
Playing the devils advocate, I would suggest not to push route deletion to the space repository because the space repository would have to either depend on the route repo, or know how to handle routes (which is responsibility of the route repo). Handlers typically delegate to repositories one they handle the request parameters, therefore delegating to different repositories kind of make sense to me.

Reply:
A couple of days later, when looking at this PR again, I support @georgethebeatle suggestion to push the logic down to the repository. But instead of trying to alter the `DeleteRoute` method (where its message specifies the route to be deleted explicitly), I would suggest creating a new method - `DeleteUnmappedRoutes(...., spaceGUID)`